compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3369,7 +3369,7 @@ generateS390CompareAndBranchOpsHelper(TR::Node * node, TR::CodeGenerator * cg, T
                if(testRegister->getKind() == TR_GPR64 || cg->use64BitRegsOn32Bit())
                   compareOpCode = isUnsignedCmp ? TR::InstOpCode::CLG : TR::InstOpCode::CG;
                else
-                  compareOpCode = isUnsignedCmp ? TR::InstOpCode::getCmpLogicalOpCodeFromNode(nonConstNode) : TR::InstOpCode::getCmpOpCodeFromNode(nonConstNode);
+                  compareOpCode = isUnsignedCmp ? TR::InstOpCode::getCmpLogicalOpCode() : TR::InstOpCode::getCmpOpCode();
```

Comment:
The reason why this change is safe is because we know at this point `constType` is a `TR::Address` or a `TR::Int64`. We also know `node` is an `TR::if*cmp*` tree which must have type safe children. This implies `nonConstType` must also be equivalent to the type of `constNode`.

As such on 64-bit we must always use a 64-bit compare (since the `constType` is a 64-bit value) and on 32-bit we must always use a 32-bit compare.